pkg/cfn/manager/api.go

```diff
@@ -178,8 +178,23 @@ func (c *StackCollection) CreateStack(ctx context.Context, stackName string, res
 
 // createClusterStack creates the cluster stack
 func (c *StackCollection) createClusterStack(ctx context.Context, stackName string, resourceSet builder.ResourceSetReader, errCh chan error) error {
+	// Do a preliminary check, so we don't end up waiting for nothing.
+	stack, err := c.DescribeStack(ctx, &types.Stack{
+		StackName: &stackName,
+	})
+	if err != nil && !IsStackDoesNotExistError(err) {
+		return err
+	}
+	if stack != nil {
+		switch stack.StackStatus {
+		case types.StackStatusCreateComplete, types.StackStatusDeleteComplete:
+			close(errCh) // no need to wait on this task anymore
+			return nil
+		}
+	}
```

Comment:
That would happen at the createStackCall a few calls back. This is in the Wait part of the chain.

Reply:
ah, yes, it appears so

Reply:
I removed this call, but for the record, sadly, because of how the wait works:
```go
		stack, err := waiter.WaitForStack(ctx, c.cloudformationAPI, *stack.StackId, *stack.StackName, func(attempts int) time.Duration {
			// Wait 30s for the first two requests, and 1m for subsequent requests.
			if attempts <= 2 {
				return 30 * time.Second
			}
			return 1 * time.Minute
		})
```

This will make the unit test run for at least 30 seconds. ¯\\\_(ツ)\_/¯

Reply:
Could skip the initial 30s wait and add to the retries.

Reply:
Technically it doesn't make sense since we are waiting on an API call, which will not even be registered in a second. :D
Also, if you mean attempts, that's the parameter coming from WaitForStack. :) Or did you mean to track the retries externally?